lib/faraday/http_cache/response.rb

```diff
@@ -38,7 +38,11 @@ def initialize(payload = {})
         @now = Time.now
         @payload = payload
         wrap_headers!
-        headers['Date'] ||= @now.httpdate
+        begin
```

Comment:
the goal if this `begin` + `rescue` block is to do something on the lines of "Try to parse the `Date` header, if it fails set it to `@now`, right? I think it's worth to push it to a private method like `ensure_date_header` so we know that it's a required step on processing a response on our side.